### PR TITLE
deploy chroot scripts

### PR DESCRIPTION
**Pull request by cryslith**
Add scripts for creating a deploy chroot.  This allows for multiple installations of spire on the same machine, and thus for running multiple autodeploys at once.

See #363.

---

Checklist:

 - [x] I have split up this change into one or more appropriately-delineated commits.
 - [x] The first line of each commit is of the form "[component]: do something"
 - [x] I have written a complete, multi-line commit message for each commit.
 - [x] I have formatted any Go code that I have changed with gofmt.
 - [x] I have signed each commit with my GPG key.
 - [ ] My changes have passed CI.
 - [x] I have built my changes locally, and tested that the changes work as expected.
 - [x] I have deployed a cluster incorporating my changes, and checked that it deploys successfully.
 - [x] I have assigned this issue to an appropriate reviewer. (Choose @celskeggs if you are not otherwise certain.)
 - [x] I consider my PR complete and ready to be merged without my further input, assuming that it passes CI and code review.

### CODE REVIEW

File: deploy-chroot/create.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+fi
+
+if [ "$USER" = "root" ]
+then
+    echo "this script should be run as a user with sudo capabilities" 1>&2
+    exit 1
+fi
+
+mkdir -m 'u=rwx,go=rx' "${HOMEWORLD_DEPLOY_CHROOT}"
+# todo: separate chroot-packages; install bridge-utils; qemu-utils; linux-image-amd64; qemu-system-x86; whichever gives ssh-agent
```
Comment (celskeggs) [inline deploy-chroot/create.sh]: Any reason not to do this now, if we need it anyway?

File: deploy-chroot/reinstall-spire.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+if [ "${HOMEWORLD_DEPLOY_CHROOT:-}" = "" -o ! -e "${HOMEWORLD_DEPLOY_CHROOT}" ]
+then
+    echo "invalid path to chroot: ${HOMEWORLD_DEPLOY_CHROOT:-}" 1>&2
+    echo '(have you populated $HOMEWORLD_DEPLOY_CHROOT?)'
+    echo '(have you created a chroot?)'
+    exit 1
+fi
+
+ORIG_PWD="$(pwd)"
```
Comment (celskeggs) [inline deploy-chroot/reinstall-spire.sh]: Do you need ORIG_PWD?

File: platform/extract-apt-setup.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -e -u
+
+cd "$(dirname "$0")"
```
Comment (krawthekrow) [inline platform/extract-apt-setup.sh]: If this is always run in the deploy chroot, you could just use absolute paths. Also, if this is a utility script, why not add `bazel build //apt-setup:package.deb` to it too?

Comment (cryslith) [reply platform/extract-apt-setup.sh]: this is run in the build chroot.  I guess I could add that

File: deploy-chroot/generate-setup.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+if __name__ == '__main__':
+    p = argparse.ArgumentParser()
+    p.add_argument('subnet_byte', help='A unique-per-host byte for which private network to use.')
+    args = p.parse_args()
+
+    if not (0 <= int(args.subnet_byte) < 256):
+        raise ValueError('subnet_byte must be a byte')
+
+    with open(os.path.expandvars('$HOMEWORLD_DIR/setup.yaml.in')) as fin, open(os.path.expandvars('$HOMEWORLD_DIR/setup.yaml'), 'w') as fout:
+        fout.write(fin.read().format(x=args.subnet_byte))
```
Comment (celskeggs) [inline deploy-chroot/generate-setup.py]: Can we have a quick overview of how to use these new scripts in the documentation somewhere?

File: deploy-chroot/reinstall-spire.sh (hunk 2 above)
Comment (cryslith) [reply deploy-chroot/reinstall-spire.sh]: oops

File: deploy-chroot/create.sh (hunk 1 above)
Comment (cryslith) [reply deploy-chroot/create.sh]: oops, that todo is todone

File: deploy-chroot/generate-setup.py (hunk 4 above)
Comment (cryslith) [reply deploy-chroot/generate-setup.py]: yes

File: deploy-chroot/reinstall-spire.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+cd "$(dirname "$0")"
+cd "$(git rev-parse --show-toplevel)"
+sudo systemd-nspawn -E PATH="/usr/local/bin:/usr/bin:/bin" -M "$(basename $HOMEWORLD_DEPLOY_CHROOT)" --bind $(pwd):/homeworld:norbind -u "$USER" -a -D "$HOMEWORLD_DEPLOY_CHROOT" bash -c "cd /homeworld && (sudo apt-get -y purge homeworld-apt-setup && sudo apt-get -y remove 'homeworld-*' || true) && sudo dpkg -i apt-setup.deb && sudo apt-get update && sudo apt-get -y install homeworld-spire"
```
Comment (krawthekrow) [inline deploy-chroot/reinstall-spire.sh]: This is missing an apt-get clean.

File: deploy-chroot/generate-setup.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+if __name__ == '__main__':
+    p = argparse.ArgumentParser()
+    p.add_argument('subnet_byte', help='A unique-per-host byte for which private network to use.')
+    args = p.parse_args()
+
+    if not (0 <= int(args.subnet_byte) < 256):
+        raise ValueError('subnet_byte must be a byte')
+
+    with open(os.path.expandvars('$HOMEWORLD_DIR/setup.yaml.in')) as fin, open(os.path.expandvars('$HOMEWORLD_DIR/setup.yaml'), 'w') as fout:
```
Comment (krawthekrow) [inline deploy-chroot/generate-setup.py]: `setup.yaml.in` doesn't seem to be autogenerated/provided. Add that to the documentation somewhere?

File: deploy-chroot/create.sh (hunk 1 above)
Comment (krawthekrow) [inline deploy-chroot/create.sh]: I think this todo has already been done?

Comment (cryslith) [reply deploy-chroot/create.sh]: fixed

File: deploy-chroot/generate-setup.py (hunk 6 above)
Comment (cryslith) [reply deploy-chroot/generate-setup.py]: added

File: deploy-chroot/reinstall-spire.sh (hunk 5 above)
Comment (cryslith) [reply deploy-chroot/reinstall-spire.sh]: fixed

File: platform/extract-apt-setup.sh (hunk 3 above)
Comment (cryslith) [reply platform/extract-apt-setup.sh]: added

**Pull request merged**